### PR TITLE
Improvement for headers authentication:

### PR DESCRIPTION
**Pull request by jdrossl**
- Support custom headers names

https://github.com/craftercms/craftercms/issues/4603

### CODE REVIEW

**Review by avasquez614 — CHANGES_REQUESTED** [review]
Can we also make sure that the bean properties are wired to properties defined in `server-config.properties`?

File: src/main/resources/crafter/engine/server-config.properties [hunk 1]
```diff
@@ -242,6 +242,17 @@ crafter.security.login.url=/crafter-security-login
 crafter.security.logout.url=/crafter-security-logout
 # The value for the security token used in header authentication
 crafter.security.header.token=my_secure_token
+# The prefix used for all header names
+crafter.security.header.names.prefix=MELLON_
```
Comment (sumerjabri) [inline src/main/resources/crafter/engine/server-config.properties:246]: @avasquez614 @jdrossl is there a reason for this to continue to be defaulted to `MELLON_`?

Comment (avasquez614) [reply src/main/resources/crafter/engine/server-config.properties:246]: I think it's for backward compatibility no? I wonder so myself, I think we should remove it if possible.

Comment (jdrossl) [reply src/main/resources/crafter/engine/server-config.properties:246]: yes it's for backward compatibility, we can change it and document it in the release notes

Comment (sumerjabri) [reply src/main/resources/crafter/engine/server-config.properties:246]: @jdrossl remove the MELLON_ prefix and replace it with CRAFTER_ here and in the docs.
Thanks.

Comment (sumerjabri) [reply src/main/resources/crafter/engine/server-config.properties:246]: @alhambrav please include this change in the release notes.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
See notes.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**